bmc/engine/verify_settling_time.h

```diff
@@ -0,0 +1,1436 @@
+
+#define v_get_val(x, dim, i) (v_chk_idx(x, dim, i) ? (x)[(i)] : (0))
+
+#define	m_chk_idx(A, A_m, A_n, i, j) ((i)>=0 && (i)<A_m && (j)>=0 && (j)<=A_n)
+
+#define	m_get_val(A, A_m, A_n, i, j) (m_chk_idx(A, A_m, A_n, i, j) ? \
+    (A)[(i)][(j)] : (0))
+
+#define	m_entry(A, A_m, A_n, i, j) m_get_val(A, A_m, A_n, i, j)
+
+#ifdef DEBUG
+#define printfc(c) printf("%f%c%fi\n", c.real, (c.imag>=0.0f)? '+':'\0', c.imag)
+#endif
+
+/* type independent min and max operations */
+#ifndef max
+#define	max(a, b) ((a) > (b) ? (a) : (b))
+#endif /* max */
+#ifndef min
+#define	min(a, b) ((a) > (b) ? (b) : (a))
+#endif /* min */
+
+/* complex number definition */
+typedef	struct
+{
+  double real, imag;
+}
+CMPLX;
+
+/* peak value definition */
+typedef	struct
+{
+  double mp;
+  int kp;
+}
+PKVL;
+
+/* last state vector definition */
+typedef	struct
+{
+  double xk[MAX_SIZE][MAX_SIZE];
+  int lastState;
+}
+LST;
+
+/* Global variables */
+// xk stores the last computed state of the system
+LST xk;
+
+/******************************math.h Functions******************************/
+
+/* sp_fabs -- absolute value of floating-point number */
+double sp_fabs(double n)
+{
+  if(n >= 0)
+    return n; //if positive, return without ant change
+  else
+    return (-n); //if negative, return a positive version
+}
+
+/* sp_ceil -- the smallest integer value greater than or equal to x */
+double sp_ceil(double x)
+{
+  union
+  {
+    float f;
+    int i;
+  }float_int;
+  float_int.f = x;
+  // Extract sign, exponent and mantissa
+  // Bias is removed from exponent
+  int sign=float_int.i >> 31;
+  int exponent=((float_int.i & 0x7fffffff) >> 23) - 127;
+  int mantissa=float_int.i & 0x7fffff;
+  // Is the exponent less than zero?
+  if(exponent < 0)
+  {
+    // In this case, x is in the open interval (-1, 1)
+    if(x <= 0.0f)
+      return 0.0f;
+    else
+      return 1.0f;
+  }
+  else
+  {
+    // Construct a bit mask that will mask off the
+    // fractional part of the mantissa
+    int mask = 0x7fffff >> exponent;
+    // Is x already an integer (i.e. are all the
+    // fractional bits zero?)
+    if((mantissa & mask) == 0)
+      return x;
+    else
+    {
+      // If x is positive, we need to add 1 to it
+      // before clearing the fractional bits
+      if(!sign)
+      {
+        mantissa += 1 << (23-exponent);
+
+        // Did the mantissa overflow?
+        if(mantissa & 0x800000)
+        {
+          // The mantissa can only overflow if all the
+          // integer bits were previously 1 -- so we can
+          // just clear out the mantissa and increment
+          // the exponent
+          mantissa = 0;
+          exponent++;
+        }
+      }
+      // Clear the fractional bits
+      mantissa &= ~mask;
+    }
+  }
+  // Put sign, exponent and mantissa together again
+  float_int.i = (sign << 31) | ((exponent+127) << 23) | mantissa;
+  return (double)float_int.f;
+}
+
+/* sp_pow -- returns a raised to the power of n i.e. a^n */
```

Comment:
Can you please add comments to all functions using the doxygen style?

For example:

```
/**
 * Converts a single precision float to fixed point representation
 * @param f to float number to be converted
 * @return fixed point representation of input.
 */
```

https://github.com/ssvlab/dsverifier/blob/master/bmc/core/fixed_point.h#L3541